### ISSUE TITLE
Department members not receiving notifications when issue transferred

### ISSUE THREAD

**Issue by marcobarbaro** [osTicket/osTicket#5073]
### Description

When transferring tickets to a single department agent the notification is received, instead when transferring the ticket to a specific department the members do not receive notifications

### Steps to Reproduce

1. Create an issue
2. Do enable notifications to department members when transferring from general settings
3. Transfer the issue from a department to another

**Expected behavior:** 

Some mail should be received from each of the selected department members

**Actual behavior:** 

Notifications are not received

### Versions

osTicket Version | 1.12.2 (develop version tested too)
Web Server Software | Apache/2.4.41 (Win32) OpenSSL/1.1.1c PHP/7.3.10
MySQL Version | 10.1.37 (MariaDB)
PHP Version | 7.3.10

Windows Server 2016

Client operative system Windows 7, Broser is Google Chrome 77.0.3865.90

**Comment by JediKev** [comment]
@marcobarbaro

There are a few things to check in this case:
1.) Are the Transfer Alerts Enabled? (**Admin Panel > Settings > Tickets > Alerts and Notices > Ticket Transfer Alert > Enable > Save Changes**)
2.) Are the Alerts being sent to Primary Members and Extended Access Members, only Primary Members, only Admin Email, or nobody? (**Admin Panel > Agents > Departments > choose Department > Settings > Alerts and Notices > Recipients**)
3.) Is the Ticket Assigned? If so it will not alert Department Members, only the assigned Agent/Team will receive the Alerts:
- https://github.com/osTicket/osTicket/blob/d7120e97371246744eb7dea532a7dbd7e15b6669/include/class.ticket.php#L2352

Cheers.

**Comment by marcobarbaro** [comment]
Hi @JediKev 
thanks for your reply.
About points 1) and 2) it was all enabled but I didn't know that, if the ticket has been assigned once, department members would not receive the alerts.
So I tried creating a new ticket without assigning it to an agent and then ticket transfer notifications worked as expected!
However there are still two points I'd like to discuss with you.
First I was wondering if this behaviour could be modified, because notifying members too could be useful.
Second I noticed doing tests that tickets transferring has become slow and sometimes the loading windows does not vanish even after minutes! This behaviour happens also on other clients, not only from mine. I noticed it after upgrading to PHP 7.3.x last week, as suggested by in the forum by @ntozier because before that we had PHP 5.6 with OsTicket 1.12.2 and PHP 5.6 is not supported.
Moreover this behaviour happens also into in a test environment where I installed OsTicket 1.12.2 develop branch.
What we saw - from various browsers (Mozilla Firefox 69 and Chrome 77) - is that if you click browser back button during "loading", the ticket has become transferred as expected.
Could you please help me understand if this could be a bug?
Thanks
Marco

**Comment by JediKev** [comment]
@marcobarbaro

>First I was wondering if this behaviour could be modified, because notifying members too could be useful.

Sure just modify the `if()`/`elseif` statements to your liking.

>Second I noticed doing tests that tickets transferring has become slow and sometimes the loading windows does not vanish even after minutes! This behaviour happens also on other clients, not only from mine. I noticed it after upgrading to PHP 7.3.x last week, as suggested by in the forum by @ntozier because before that we had PHP 5.6 with OsTicket 1.12.2 and PHP 5.6 is not supported.

We have not noticed any slowness issues when transferring etc. You will have to check your sever performance statistics and do some general debugging to figure out where the issue stems from. If it stems from osTicket we will take a look but like I said we haven’t noticed any slowness issues when transferrring with v1.12.2. I would also try downgrading to PHP 5.6-7.2 to see if this improves performance as it might be a PHP 7.3 related issue.

Cheers.

**Comment by marcobarbaro** [comment]
Hi @JediKev ,
from a first test it seems that downgrading to PHP 7.2 today, in a test, environment, makes return faster the ticket transferring process between departments.
Today I'll bring PHP 7.2 to production and see if there it works too...

**Comment by marcobarbaro** [comment]
Hi @JediKev ,
the question about "loading" window stuck when transferring ticket between departments is still confirmed even after downgrading production to php 7.2.

We have about 100-150 quantity of opened ticket in one of the two departments in which we are doing tests , perhaps this info could be useful?
I'm asking because server performance seems to be good. 
CPU utilization is below 10%
Memory utilization is about 7GB on 16GB available.
Apache and DB (Maria DB) are on the same Windows server 2016 VM.

Thanks
Marco

**Comment by marcobarbaro** [comment]
Hi @JediKev a quick update.
It seems that enabling "Department Members" and "Department Manager" notifications on Department transfer, as I did at the beginning of this issue, makes the transfer process slow.
Perhaps because osticket code has to cycle on all the department members, even if the ticket has been already assigned to a user?
Is this behaviour ovveridable?
Thanks
Marco

**Comment by JediKev** [comment]
@marcobarbaro 

>Perhaps because osticket code has to cycle on all the department members, even if the ticket has been already assigned to a user?

No, it will check for assignment first and only add those members without cycling through members:
- https://github.com/osTicket/osTicket/blob/d7120e97371246744eb7dea532a7dbd7e15b6669/include/class.ticket.php#L2341-L2350

Cheers.

**Comment by marcobarbaro** [comment]
Thanks but how it could be in your opinion that after disabling "department members" flag the transfer has become quick as it was before?
Mail settings are to use smtp settings of the account configured, not php email function.

**Comment by marcobarbaro** [comment]
Hi @JediKev I confirm what I told you last week, after disabling department members option the loading windows become faster again... what do you think about it?
Thanks
Marco

**Comment by JediKev** [comment]
@marcobarbaro 

What do I think about it? I think it doesn't make sense, since the code isn't supposed to touch that setting/alert if the Ticket is Assigned. We will have to do some extensive testing as really this should not happen.

Cheers.

**Comment by marcobarbaro** [comment]
Thanks for you reply @JediKev , in the meantime I think I will not face the issue again, so you could close this issue as the original matter has been solved by your hint.
Marco

**Comment by marcelpinheiro** [comment]
If you automatic assign to a team, the alert will not be sent. Set only the department
